LoopUp.Siesta/Patch/JsonPatchDocumentHelpers.cs

```diff
@@ -0,0 +1,193 @@
+namespace LoopUp.Siesta.Patch
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using LoopUp.Siesta.Extensions;
+    using Microsoft.AspNetCore.JsonPatch;
+
+    /// <summary>
+    /// Helper class for making and working with JsonPatchDocuments.
+    /// </summary>
+    public static class JsonPatchDocumentHelpers
+    {
+        /// <summary>
+        /// Generate a JsonPatchDocument through the comparison of two versions of the same class.
+        /// </summary>
+        /// <param name="original">Original version of the class.</param>
+        /// <param name="modified">New version of the class.</param>
+        /// <typeparam name="T">The type of the class.</typeparam>
+        /// <returns>A JsonPatchDocument for the two objects.</returns>
+        public static JsonPatchDocument GeneratePatchDocument<T>(T original, T modified)
+            where T : class
+        {
+            var patchDocument = new JsonPatchDocument();
+
+            PopulatePatchForObject(original, modified, patchDocument, string.Empty);
+
+            return patchDocument;
+        }
+
+        /// <summary>
+        /// Recursive method to populate the Patch document for 2 objects.
+        /// </summary>
+        /// <param name="originalObject">The original version of the object.</param>
+        /// <param name="modifiedObject">The modified version of the object.</param>
+        /// <param name="patchDocument">The PATCH document to populate.</param>
+        /// <param name="path">The path to use for PATCH document paths.</param>
+        /// <typeparam name="T">The object type.</typeparam>
+        private static void PopulatePatchForObject<T>(T? originalObject, T? modifiedObject, JsonPatchDocument patchDocument, string path)
+            where T : class
+        {
+            // Use whichever object is no null to get the properties on the object
```

Comment:
💦 not* null?

Reply:
If there are no functional changes that need making, I will make all these types of changes in a subsequent PR as it is probs gonna cause me some merge conflicts with later PRs 👍